File: src/main/java/com/pengrad/telegrambot/model/Location.java
package com.pengrad.telegrambot.model;

/**
 * stas
 * 8/5/15.
 */
public class Location {

    private Float longitude;
    private Float latitude;

    Location() {
    }

    public Float longitude() {
        return longitude;
    }

    public Float latitude() {
        return latitude;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Location location = (Location) o;

        if (longitude != null ? !longitude.equals(location.longitude) : location.longitude != null) return false;
        return latitude != null ? latitude.equals(location.latitude) : location.latitude == null;
    }

    @Override
    public int hashCode() {
        int result = longitude.hashCode();
        result = 31 * result + latitude.hashCode();
        return result;
    }

    @Override
    public String toString() {
        return "Location{" +
                "longitude=" + longitude +
                ", latitude=" + latitude +
                '}';
    }
}

File: core/src/commonMain/kotlin/com/pengrad/telegrambot/model/Location.kt
package com.pengrad.telegrambot.model

import java.io.Serializable

/**
 * stas
 * 8/5/15.
 */
class Location : Serializable {

    private val longitude: Float? = null
    private val latitude: Float? = null

    fun longitude(): Float? {
        return longitude
    }

    fun latitude(): Float? {
        return latitude
    }

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false

        val location = o as Location?

        if (if (longitude != null) longitude != location!!.longitude else location!!.longitude != null) return false
        return if (latitude != null) latitude == location.latitude else location.latitude == null
    }

    override fun hashCode(): Int {
        var result = longitude?.hashCode() ?: 0
        result = 31 * result + (latitude?.hashCode() ?: 0)
        return result
    }

    override fun toString(): String {
        return "Location{" +
                "longitude=" + longitude +
                ", latitude=" + latitude +
                '}'.toString()
    }

    companion object {
        private const val serialVersionUID = 0L
    }
}
